### PR TITLE
Feature/304 objective dialog

### PR DESCRIPTION
**Pull request by lkleisa**

### CODE REVIEW

**Review by MarcEgliP — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.component.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
       position="end"
     >
       <app-drawer-content
-        *ngIf="sidenavContentInformation != undefined"
+        *ngIf="sidenavContentInformation != undefined && drawerOpen"
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply frontend/src/app/app.component.html]: done

**Review by MarcEgliP — APPROVED** [review]

File: backend/src/main/java/ch/puzzle/okr/repository/CheckInRepository.java [hunk 2]
```diff
@@ -9,4 +9,6 @@ public interface CheckInRepository extends CrudRepository<CheckIn, Long> {
     List<CheckIn> findCheckInsByKeyResultIdOrderByCreatedOnDesc(Long keyResultId);
 
     CheckIn findFirstByKeyResultIdOrderByCreatedOnDesc(Long keyResultId);
+
```
Comment (peggimann) [inline backend/src/main/java/ch/puzzle/okr/repository/CheckInRepository.java:12]: is not nescessary

File: backend/src/main/java/ch/puzzle/okr/service/validation/ObjectiveValidationService.java [hunk 3]
```diff
@@ -36,4 +41,13 @@ public void validateOnUpdate(Long id, Objective model) {
         doesEntityExist(id);
         validate(model);
     }
+
+    @Override
+    public void validateOnDelete(Long id) {
+        throwExceptionWhenIdIsNull(id);
+        doesEntityExist(id);
+        if (checkInBusinessService.getCheckInAmountByObjectiveId(id) != 0) {
```
Comment (peggimann) [inline backend/src/main/java/ch/puzzle/okr/service/validation/ObjectiveValidationService.java]: this error should not be thrown, a delete on an objective should delete its corresponding key results and the corresponding check ins

File: backend/src/test/java/ch/puzzle/okr/service/business/CheckInBusinessServiceTest.java [hunk 4]
```diff
@@ -154,4 +154,10 @@ void shouldGetLastCheckInByKeyResultId() {
         checkInBusinessService.getLastCheckInByKeyResultId(this.ordinalKeyResult.getId());
         verify(checkInPersistenceService, times(1)).getLastCheckInOfKeyResult(this.ordinalKeyResult.getId());
     }
+
+    @Test
+    void shouldCallGetCheckInAmountByObjectiveId() {
+        checkInBusinessService.getCheckInAmountByObjectiveId(1L);
```
Comment (peggimann) [inline backend/src/test/java/ch/puzzle/okr/service/business/CheckInBusinessServiceTest.java:160]: this test can be removed

File: backend/src/test/java/ch/puzzle/okr/service/validation/ObjectiveValidationServiceTest.java [hunk 5]
```diff
@@ -272,4 +276,12 @@ void validateOnDelete_ShouldThrowExceptionIfObjectiveIdIsNull() {
         assertEquals("Id is null", exception.getReason());
     }
 
+    @Test
+    void validateOnDelete_ShouldThrowExceptionIfObjectiveHasCheckIns() {
+        when(checkInBusinessService.getCheckInAmountByObjectiveId(any())).thenReturn(2);
+        ResponseStatusException exception = assertThrows(ResponseStatusException.class,
```
Comment (peggimann) [inline backend/src/test/java/ch/puzzle/okr/service/validation/ObjectiveValidationServiceTest.java:282]: this test is not needed

File: backend/src/test/java/ch/puzzle/okr/service/validation/ObjectiveValidationServiceTest.java [hunk 6]
```diff
@@ -38,6 +39,9 @@ class ObjectiveValidationServiceTest {
     Quarter quarter;
     Team team;
 
+    @MockBean
+    CheckInBusinessService checkInBusinessService = Mockito.mock(CheckInBusinessService.class);
```
Comment (peggimann) [inline backend/src/test/java/ch/puzzle/okr/service/validation/ObjectiveValidationServiceTest.java:43]: this mock is not needed

File: frontend/src/app/app.component.ts [hunk 7]
```diff
@@ -80,11 +80,14 @@ export class AppComponent implements OnInit, OnDestroy {
 
   openDrawer() {
     this.drawerOpen = true;
+    document.getElementById('bannerComponent')!.classList.add('backdrop');
```
Comment (peggimann) [inline frontend/src/app/app.component.ts:83]: should not bee needed anymore.

**Review by peggimann — CHANGES_REQUESTED** [review]
include tests for services

File: backend/src/main/java/ch/puzzle/okr/repository/CheckInRepository.java (hunk 2 above)
Comment (MarcEgliP) [reply backend/src/main/java/ch/puzzle/okr/repository/CheckInRepository.java:12]: Still needed

**Review by peggimann — APPROVED** [review]

**Pull request merged**